arangod/ClusterEngine/ClusterCollection.cpp

```diff
@@ -0,0 +1,532 @@
+  // objectId might be undefined on the coordinator
+  TRI_ASSERT(result.isOpenObject());
+
+  if (_engineType == ClusterEngineType::MMFilesEngine) {
+    result.add("doCompact", VPackValue(Helper::readBooleanValue(
+                                _info.slice(), "doCompact", true)));
+    result.add("indexBuckets", VPackValue(Helper::readNumericValue(
+                                   _info.slice(), "indexBuckets",
+                                   MMFilesCollection::defaultIndexBuckets)));
+    result.add("isVolatile", VPackValue(Helper::readBooleanValue(
+                                 _info.slice(), "isVolatile", false)));
+    result.add("journalSize",
+               VPackValue(Helper::readNumericValue(_info.slice(), "journalSize",
+                                                   TRI_JOURNAL_DEFAULT_SIZE)));
+
+  } else if (_engineType == ClusterEngineType::RocksDBEngine) {
+    result.add("cacheEnabled", VPackValue(Helper::readBooleanValue(
+                                   _info.slice(), "cacheEnabled", false)));
+
+  } else {
+    TRI_ASSERT(false);
+    THROW_ARANGO_EXCEPTION(TRI_ERROR_INTERNAL);
+  }
+}
+
+/// @brief return the figures for a collection
+std::shared_ptr<VPackBuilder> ClusterCollection::figures() {
+  auto builder = std::make_shared<VPackBuilder>();
+  builder->openObject();
+  builder->close();
+
+  int res =
+      figuresOnCoordinator(_logicalCollection->vocbase().name(),
+                           std::to_string(_logicalCollection->id()), builder);
+  if (res != TRI_ERROR_NO_ERROR) {
+    THROW_ARANGO_EXCEPTION(res);
+  }
+  return builder;
+}
+
+void ClusterCollection::figuresSpecific(
+    std::shared_ptr<arangodb::velocypack::Builder>& builder) {
+  THROW_ARANGO_EXCEPTION(TRI_ERROR_NOT_IMPLEMENTED);  // not used here
+}
+
+/// @brief closes an open collection
+int ClusterCollection::close() {
+  READ_LOCKER(guard, _indexesLock);
+  for (auto it : _indexes) {
+    it->unload();
+  }
+  return TRI_ERROR_NO_ERROR;
+}
+
+void ClusterCollection::load() {
+  READ_LOCKER(guard, _indexesLock);
+  for (auto it : _indexes) {
+    it->load();
+  }
+}
+
+void ClusterCollection::unload() {
+  READ_LOCKER(guard, _indexesLock);
+  for (auto it : _indexes) {
+    it->unload();
+  }
+}
+
+TRI_voc_rid_t ClusterCollection::revision(transaction::Methods* trx) const {
+  THROW_ARANGO_EXCEPTION(TRI_ERROR_NOT_IMPLEMENTED);
+}
+
+uint64_t ClusterCollection::numberDocuments(transaction::Methods* trx) const {
+  THROW_ARANGO_EXCEPTION(TRI_ERROR_NOT_IMPLEMENTED);
+}
+
+/// @brief report extra memory used by indexes etc.
+size_t ClusterCollection::memory() const { return 0; }
+
+void ClusterCollection::open(bool ignoreErrors) {}
+
+void ClusterCollection::prepareIndexes(
+    arangodb::velocypack::Slice indexesSlice) {
+  WRITE_LOCKER(guard, _indexesLock);
+  TRI_ASSERT(indexesSlice.isArray());
+
+  StorageEngine* engine = EngineSelectorFeature::ENGINE;
+  std::vector<std::shared_ptr<Index>> indexes;
```

Comment:
actually it would be nice to have TRI_ASSERT everywhere where EngineSelectorFeature::ENGINE is used since the check AFAIK is free in release but in tests it helps to know were static variable were not initialized